Model/RefundSuccessfulProcessor.php

```diff
@@ -47,21 +49,18 @@ public function __construct(
      */
     public function processSuccessfulWebhook($params, $jobName)
     {
-        $data = $params[0];
-        $json = !empty($data) ? json_decode($data, true) : [];
+        $data = $params;
+        $json = !empty($data) ? $data : [];
         $refundId = isset($json['data']['refundId']) ? $json['data']['refundId'] : '';
         if (!$refundId) {
             throw new LocalizedException(new Phrase("Refund ID Not Present!"));
         }
         $memoId = $this->collection->addFieldToFilter('refund_id', ['eq' => $refundId])
             ->getFirstItem()->getCreditMemoId();
         if ($memoId) {
-            $creditmemoModel = $this->creditmemoFactory->create();
-            $creditmemoModel->setData([
-                'entity_id' => $memoId,
-                'creditmemo_status' => 2
-            ]);
-            $creditmemoModel->save();
+            $memoData = $this->creditmemoRepository->get($memoId);
+            $memoData->addData(['state' => 2]);
```

Comment:
can we represent state in textual value?

Reply:
Updated